apps/test/unit/code-studio/pd/application/TeacherApplicationTest.js

```diff
@@ -0,0 +1,139 @@
+    const page = mount(<TeacherApplication {...defaultProps} />);
+    expect(page.find('SchoolAutocompleteDropdown').prop('value')).to.equal(
+      undefined
+    );
+  });
+  it('Sets the school dropdown value from props', () => {
+    const page = mount(<TeacherApplication {...defaultProps} schoolId="50" />);
+    expect(page.find('SchoolAutocompleteDropdown').prop('value')).to.equal(
+      '50'
+    );
+  });
+  it('Sets the school dropdown value from storage', () => {
+    window.sessionStorage.setItem(
+      'TeacherApplication',
+      JSON.stringify({data: {school: '25'}})
+    );
+    const page = mount(<TeacherApplication {...defaultProps} />);
+    expect(page.find('SchoolAutocompleteDropdown').prop('value')).to.equal(
+      '25'
+    );
+  });
+  it('Logs user id on initialization', () => {
+    const fc = sinon.stub(firehoseClient, 'putRecord');
+    shallow(<TeacherApplication {...defaultProps} />);
+    expect(fc.calledWith(sinon.match({userId: 1})));
+  });
+  it('Reports to google analytics', () => {
+    const form = mount(<TeacherApplication {...defaultProps} />);
+    const nextButton = form.find('#next').first();
+    nextButton.simulate('click');
+    expect(ga.getCall(2).calledWith());
+  });
+
+  describe('getInitialData', () => {
+    const savedFormData =
+      '{' +
+      '"firstName": "Brilliant", ' +
+      '"previousPd": ["CS Principles","CS Discoveries"]' +
+      '}';
+    const parsedData = JSON.parse(savedFormData);
+    const schoolId = '5';
+    const teacherApplication = isolateComponent(
+      <TeacherApplication {...defaultProps} allowPartialSaving />
+    );
+
+    it('has no initial data if there is nothing in session storage, no school id, and no form data', () => {
+      expect(
+        teacherApplication.findOne('FormController').props.getInitialData()
+      ).to.deep.equal({});
+    });
+    it('has saved form data if nothing in session storage and no school id', () => {
+      teacherApplication.mergeProps({savedFormData});
+      expect(
+        teacherApplication.findOne('FormController').props.getInitialData()
+      ).to.deep.equal(parsedData);
+    });
+    it('has saved form data and school id if nothing in session storage', () => {
+      teacherApplication.mergeProps({savedFormData, schoolId});
+      expect(
+        teacherApplication.findOne('FormController').props.getInitialData()
+      ).to.deep.equal({
+        ...parsedData,
+        school: schoolId
+      });
+    });
+    it('overwrites existing school id if the saved form data has school id', () => {
+      teacherApplication.mergeProps({
+        savedFormData: `{"school": "16"}`,
+        schoolId
+      });
+      expect(
+        teacherApplication.findOne('FormController').props.getInitialData()
+      ).to.deep.equal({school: '16'});
+    });
+    it('has only saved form data and no school id if session storage has school info', () => {
+      window.sessionStorage.setItem(
```

Comment:
I'm not sure this is safe to use during unit tests. I would recommend using `FakeStorage` instead. OK to do in a next PR, since this one is getting big and this seems to at least be working in one existing test. this has the potential to lead to weird flakiness problems if not addressed soon though.

Reply:
Got to know. I'll make a note of this and the allowPartialSaving refactors for the next PR, but go ahead and merge thisone